README.md

```diff
@@ -56,11 +61,23 @@ Stacktrace:
 
 ## Usage
 
+### Assignment Syntax
 ``` julia
 @match pattern = value
 ```
 
-If value matches pattern, binds variables and returns `value`. Otherwise, throws MatchFailure.
+If value matches pattern, binds variables and returns `value`. Otherwise, throws `MatchFailure`.
+
+After evaluation, any variables defined within `pattern` will be bound as new variables in the enclosing scope. e.g.:
```

Comment:
'defined' is maybe confusing? perhaps 'used'? They're not necessarily new variables, right?

Reply:
Updated in latest commit. thanks!